src/libs/Navigation/NavigationRoot.js

```diff
@@ -0,0 +1,135 @@
+import _ from 'underscore';
+import React, {Component} from 'react';
+import {Linking} from 'react-native';
+import PropTypes from 'prop-types';
+import {
+    getStateFromPath,
+    getPathFromState,
+    NavigationContainer,
+} from '@react-navigation/native';
+import Onyx from 'react-native-onyx';
+import {navigationRef} from './Navigation';
+import linkingConfig from './linkingConfig';
+import AppNavigator from './AppNavigator';
+import modalRoutesPropTypes from './AppNavigator/modalRoutesPropTypes';
+import mainRoutesPropTypes from './AppNavigator/mainRoutesPropTypes';
+import sidebarRoutePropTypes from './AppNavigator/sidebarRoutePropTypes';
+import publicRoutePropTypes from './AppNavigator/publicRoutePropTypes';
+import getPathName from './getPathName';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+
+const propTypes = {
+    modalRoutes: modalRoutesPropTypes.isRequired,
+    mainRoutes: mainRoutesPropTypes.isRequired,
+    sidebarRoute: sidebarRoutePropTypes.isRequired,
+    publicRoutes: PropTypes.arrayOf(publicRoutePropTypes).isRequired,
+    authenticated: PropTypes.bool.isRequired,
+    currentlyViewedReportID: PropTypes.string,
+};
+
+const defaultProps = {
+    currentlyViewedReportID: null,
+};
+
+class NavigationRoot extends Component {
```

Comment:
Right now it looks like nothing prevents android back buttons, this causes users to be able to accidentally close the app when there's no more screens to pop. You can prob add a listener to this root nav to disabled the back button when on this screen, but you're also gonna want to disable it on specific screens that users shouldnt be able to go back on.

https://reactnavigation.org/docs/custom-android-back-button-handling

Reply:
@Maftalion I get that we want to override back button for some cases (i.e to pop current modal/page), but I'm not following why we would want to do this here at the NavigationRoot? 

> this causes users to be able to accidentally close the app when there's no more screens to pop

If there's no more screens to pop, then the app **should** close, am I missing something?